src/table/Schema.cc

```diff
@@ -473,13 +473,17 @@ int SchemaImpl::contains(SchemaItem<T> const & item, int flags) const {
 
 std::set<std::string> SchemaImpl::getNames(bool topOnly) const {
     std::set<std::string> result;
+    // set the correct separator for old table (nan or 0) and new tables (version > 0)
+    char separator;
+    if (getVersion() > 0) separator = '_';
+    else separator = '.';
     if (topOnly) {
         for (NameMap::const_iterator i = _names.begin(); i != _names.end(); ++i) {
-            std::size_t dot = i->first.find('.');
-            if (dot == std::string::npos) {
+            std::size_t sep = i->first.find(separator);
+            if (sep == std::string::npos) {
                 result.insert(result.end(), i->first);
             } else {
-                result.insert(result.end(), i->first.substr(0, dot));
+                result.insert(result.end(), i->first.substr(0, sep));
             }
         }
     } else {
```

Comment:
Use brackets and spacing in if/else clause.

Also clarify the comment such that it is unambiguous whether "nan or 0" refers to the table version or the separator.